### PR TITLE
[Flang] Maxloc elemental intrinsic lowering.

### PR DESCRIPTION
**Pull request by davemgreen**
This is an extension to #74828 to handle maxloc too, to keep the minloc and maxloc symmetric.

### CODE REVIEW

**Review by tblah — APPROVED** [review]
Looks good to me. Thank you for adding all of these!

File: flang/lib/Optimizer/HLFIR/Transforms/OptimizedBufferization.cpp [hunk 1]
```diff
@@ -812,50 +812,55 @@ class ReductionElementalConversion : public mlir::OpRewritePattern<Op> {
...
-    mlir::Location loc = minloc.getLoc();
-    fir::FirOpBuilder builder{rewriter, minloc.getOperation()};
+    mlir::Location loc = mloc.getLoc();
+    fir::FirOpBuilder builder{rewriter, mloc.getOperation()};
     mlir::Value resultArr = builder.createTemporary(
         loc, fir::SequenceType::get(
-                 rank, hlfir::getFortranElementType(minloc.getType())));
+                 rank, hlfir::getFortranElementType(mloc.getType())));
 
     auto init = [](fir::FirOpBuilder builder, mlir::Location loc,
```
Comment (tblah) [inline flang/lib/Optimizer/HLFIR/Transforms/OptimizedBufferization.cpp:851]: Windows CI needs isMax to be added to the capture specification

**Comment by davemgreen** [comment]
Thanks. I'm surprised this didn't come up in both compilers, I hadn't realized it would treat constexpr differently in lambda expressions.

**Comment by yi-wu-arm** [comment]
The [flang-aarch64-libcxx](https://lab.llvm.org/buildbot/#/builders/268)[7035](https://lab.llvm.org/buildbot/#/builders/268/builds/7035) build bot is failing: https://lab.llvm.org/buildbot/#/builders/268/builds/7035, but I'm not sure if that is related to this patch.

**Comment by davemgreen** [comment]
Yep that is me. I was just about to push a patch so long as it passed testing locally.

The MSVC compiler didn't like things one way, and the -Weverything build didn't like it the other :(

**Pull request merged**